src/hooks/usePads.js

```diff
@@ -0,0 +1,64 @@
+import { useState } from 'react'
+import { padsData } from '../common/padsData'
+import {
+  getPadNameByKey,
+  getKeyByName,
+  updateInArray,
+  toLowerNoWhiteSpace,
+  stopPinchZooming,
+} from '../common/utils'
+
+export default function usePad(samplePlayer) {
+  const [pads, setPads] = useState(padsData)
+  const [selectedPad, setSelectedPad] = useState(null)
+
+  function handleKeyDown(event) {
+    if (event.repeat) {
+      return
+    }
+    const key = event.key
+    samplePlayer.playSample(key)
+    setPadIsTriggered(key, true)
+    const padName = getPadNameByKey(pads, key)
+    setSelectedPad(padName)
+  }
+
+  function handleKeyUp(event) {
+    const key = event.key
+    setPadIsTriggered(key, false)
+  }
+
+  function setPadIsTriggered(key, triggerd) {
+    const index = pads.findIndex(pad => pad.key === key)
+    if (index > -1) {
```

Comment:
Is there a real chance, that the pad is not found?

Reply:
Yes, wenn the key is not matched